src/jemalloc.c

```diff
@@ -1585,7 +1585,7 @@ malloc_conf_init_helper(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS],
 							    "prof_leak_error is"
```

Comment:
The other thing I realized, is that the `if (!opt_prof_final)` branch creates an order dependency with prof_final, i.e. it will warn / abort if you specify "prof_leak_error:true,prof_final:true" (or prof_final specified in a later stage conf source). I'm also thinking, should we simply enable prof_final as well, since this is a new option?

Reply:
Riiiight, the fact that order in which they are specified matters is a problem. I missed that.

I think you suggestion is OK if we specify it in the docs. But then again, someone can explicitly turn off `prof_final` after they set `prof_leak_error`. This will be undetected and `prof_leak_error` won't work. Even if we specify that `prof_leak_error` enables `prof_final`, it's not obvious that the code only works if they both are set to `true`. Rejecting on options validation phase seems robust and correct here.

I think we should separate parsing options from their validation and malloc initialization. Currently it's unclear where to validate options. The `malloc_conf_init_helper` just proved to be pretty inconvenient for that, because you don't know all the values at the moment of validation.

Reply:
Yep I added a `malloc_conf_init_check_deps` step at the end. Kept it simple since we want to include it in the release.